### ISSUE TITLE
Interpreter crashes with if under is used incorrectly

### ISSUE THREAD

**Issue by iFreilicht** [uiua-lang/uiua#47]
This just happened to me on the current master branch.

```
The application panicked (crashed).
Message:  source slice length (1) does not match destination slice length (2)
Location: src/algorithm/dyadic.rs:1065

Run with COLORBT_SHOW_HIDDEN=1 environment variable to disable frame filtering.
Run with RUST_BACKTRACE=full to include source snippets.
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
                              ⋮ 7 frames hidden ⋮                               
 8: core::slice::<impl [T]>::copy_from_slice::len_mismatch_fail::he28e939c009d96d8
    at <unknown source file>
 9: uiua::algorithm::dyadic::<impl uiua::array::Array<T>>::untake::{{closure}}::h5e9361cfc8866fc7
    at <unknown source file>
10: uiua::algorithm::dyadic::<impl uiua::array::Array<T>>::untake::hcd3e5b469f645a78
    at <unknown source file>
11: uiua::primitive::<impl uiua::primitive::defs::Primitive>::run::h6fc3d8da369377c9
    at <unknown source file>
12: uiua::run::Uiua::exec::h15c55a0fdd63db50
    at <unknown source file>
13: uiua::run::Uiua::call_with_span::hd4cd9164130fa999
    at <unknown source file>
14: uiua::primitive::<impl uiua::primitive::defs::Primitive>::run::h6fc3d8da369377c9
    at <unknown source file>
15: uiua::run::Uiua::exec::h15c55a0fdd63db50
    at <unknown source file>
16: uiua::run::Uiua::exec_global_instrs::h781ded5870d44b97
    at <unknown source file>
17: uiua::run::Uiua::items::h133190d7a873af9f
    at <unknown source file>
18: uiua::run::Uiua::load_impl::h8f42d637ccbc264e
    at <unknown source file>
19: uiua::run::Uiua::load_file::h3ee99052fbdc9632
    at <unknown source file>
20: uiua::main::he5b6e4bfb8db3e21
    at <unknown source file>
21: std::sys_common::backtrace::__rust_begin_short_backtrace::h39b867a29983d447
    at <unknown source file>
22: _main
    at <unknown source file>
Error: The interpreter has crashed!
Hooray! You found a bug!
Please report this at http://github.com/uiua-lang/uiua/issues/new

code:
./day03.ua:4:3
[1_0 0_1]
⍉
≥÷2⧻∶/+.⊢.
⍜'↙1

   ╭─[./day03.ua:4:3]
 4 │ ⍜'↙1
   │   ─  
───╯
```

**Comment by kaikalii** [comment]
I've add a check for this in 93c5c85d0f, but I'm going to keep this open for now because I suspect some of the other underable functions may have this problem as well.

**Comment by kaikalii** [comment]
That one should be fixed in 443aeb316bddbcd99

**Comment by kaikalii** [comment]
This fix is live!
